pytorch3d/renderer/mesh/textures.py

```diff
@@ -1073,7 +1073,7 @@ def join_batch(self, textures: List["TexturesUV"]) -> "TexturesUV":
         faces_uvs_list += self.faces_uvs_list()
         verts_uvs_list += self.verts_uvs_list()
         maps_list += self.maps_list()
-        num_faces_per_mesh = self._num_faces_per_mesh
+        num_faces_per_mesh = self._num_faces_per_mesh.copy()
```

Comment:
In fact, I think TexturesAtlas has exactly the same bug. They should be fixed at the same time.

Reply:
Yes, this is a significant bug which was found when I was using Pytorch3D. Specifically, for  meshes with TextureUV, calling function `join_meshes_as_batch` will cause the origin mesh textureUV's `_num_faces_per_mesh` changed, due to the list reference attribute. Especially when calling `join_meshes_per_mesh` multiple times, this will cause OOM finally, because the original mesh textureUV's `_num_faces_per_mesh` list has became too long. I have not noticed the TextureAltas has the same bug because I have not used it. This is just a small change, so I committed it together. Can you please sort it out if needed?

Reply:
Yes. Let me sort it out. Best to remove this change from this PR still.